@here/harp-mapview/lib/VisibleTileSet.ts

```diff
@@ -194,8 +195,12 @@ export class VisibleTileSet {
     private m_ResourceComputationType: ResourceComputationType =
         ResourceComputationType.EstimationInMb;
 
-    constructor(private readonly camera: THREE.PerspectiveCamera, options: VisibleTileSetOptions) {
-        this.m_mapTileCuller = new MapTileCuller(camera);
+    constructor(
+        private readonly m_camera: THREE.PerspectiveCamera,
+        private readonly m_tileGeometryManager: TileGeometryManager,
```

Comment:
I find it strange that the VisibleTileSet has to know how the tile manages its geometry. Can this dependency be removed? Or does it point to a refactoring necessary in VisibleTileSet?

Reply:
I think it makes sense, because the management is important for just the visible set of tiles, and the VisibleTileSet knows best about it. The updateTiles/initTile could be moved to MapView, which would then get bigger. The Tiles should not know about the TileGeometryManager, since that may change at runtime.

Reply:
Fair enough, for now it is ok, and moving to MapView is not so nice for sure. I think in future, we should consider still to refactor this. Where the VisibleTileSet outputs just a list of visible TileKeys, and there is some other class which takes this list and then manages the creation / caching / disposing. But not a high priority.